### PR TITLE
[BI-1375] Make GIDs clickable links

### PR DESCRIPTION
**Pull request by HMS17**
# Description
**Story:** [BI-1375 - Make GIDs clickable links](https://breedinginsight.atlassian.net/browse/BI-1375)

Added pedigreeByUUID in additionalInfo for germplasm to simplify retrieval of parent UUIDs for germplasm details page links

# Dependencies
[bi-web/feature/BI-1375](https://github.com/Breeding-Insight/bi-web/pull/218)

# Testing
See testing in bi-web

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have tested my code and ensured it meets the acceptance criteria of the story
- [X] I have tested that my code works with both the brapi-java-server and BreedBase
- [X] I have create/modified unit tests to cover this change
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to documentation
- [X] I have run TAF: [_\<please include a link to TAF run>_](https://github.com/Breeding-Insight/taf/actions/runs/2173893697)

### CODE REVIEW

File: src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java [hunk 1]
```diff
@@ -157,21 +157,25 @@ private Map<String,BrAPIGermplasm> processGermplasmForDisplay(List<BrAPIGermplas
 
                 String newPedigreeString = "";
                 String namePedigreeString = "";
+                String uuidPedigreeString = "";
                 List<String> parents = Arrays.asList(germplasm.getPedigree().split("/"));
                 if (parents.size() >= 1) {
                     if (programGermplasmByFullName.containsKey(parents.get(0))) {
                         newPedigreeString = programGermplasmByFullName.get(parents.get(0)).getAccessionNumber();
                         namePedigreeString = programGermplasmByFullName.get(parents.get(0)).getDefaultDisplayName();
+                        uuidPedigreeString = programGermplasmByFullName.get(parents.get(0)).getExternalReferences().stream().filter(ref -> ref.getReferenceSource().equals(referenceSource)).map(ref -> ref.getReferenceID()).findFirst().get();
```
Comment (ctucker3) [inline src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java:166]: Can you split this into multiple lines? It is hard to read when chained on a one liner.

Comment (ctucker3) [reply src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java:166]: Also, what happens if the `filter` returns no results. Will your `findFirst` error out?

**Review by ctucker3 — CHANGES_REQUESTED** [review]

File: src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java (hunk 1 above)
Comment (HMS17) [reply src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java]: Split into multiple lines, and good point! Switched to using an orElse clause to set the value to “” if nothing is found, which leads to a GID with no link

File: src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java [hunk 2]
```diff
@@ -157,21 +157,29 @@ private Map<String,BrAPIGermplasm> processGermplasmForDisplay(List<BrAPIGermplas
...
                 String newPedigreeString = "";
                 String namePedigreeString = "";
+                String uuidPedigreeString = "";
                 List<String> parents = Arrays.asList(germplasm.getPedigree().split("/"));
                 if (parents.size() >= 1) {
                     if (programGermplasmByFullName.containsKey(parents.get(0))) {
                         newPedigreeString = programGermplasmByFullName.get(parents.get(0)).getAccessionNumber();
                         namePedigreeString = programGermplasmByFullName.get(parents.get(0)).getDefaultDisplayName();
+                        uuidPedigreeString = programGermplasmByFullName.get(parents.get(0)).getExternalReferences().
+                                stream().filter(ref -> ref.getReferenceSource().equals(referenceSource)).
```
Comment (ctucker3) [inline src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java:167]: Looks good! Did you mean to have the `.` on a separate line from the `map` and `stream`? I usually have seen it on the same line. If it's a style choice, feel free to keep it.

**Review by ctucker3 — APPROVED** [review]
Approved, one small question/suggestion.

**Pull request merged**